### ISSUE TITLE
LoginResponse has null refreshToken

### ISSUE THREAD

**Issue by simone201** [inversoft/passport-java-client#1]
I'm trying to login by using PassportClient.login method and passing a LoginRequest inside, catching the response using the non-dollar method and getting the tokens inside the response object.
While the access token and user are valid and inside the object, the access token is always NULL and i can't figure out a way to get one outside making a separare request.

Any idea what's happening?

**Comment by robotdan** [comment]
Do you mean the refresh token is null? 

For example:

```
{
  "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.e...",
  "user": {
     .... 
  }
}
```

and you're expecting

```
{
  "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.e...",
  "refreshToken": "a492#....",
  "user": {
     .... 
  }
}
```

If so, it is because you didn't provide the device information on the request. For reference, the [Login API documentation](https://www.inversoft.com/docs/passport/1.x/tech/apis/login#authenticate-a-userl).

If not, please clarify the question. Thanks.

**Comment by simone201** [comment]
Just tried what you said by adding a unique device param in the LoginRequest object (as the Java library), by generating a simple random UUID and converting it to string, seems not working anyway and still the response has null access token attribute...

**Comment by robotdan** [comment]
To clarify, are you trying to get the access token (JWT) or the refresh token? In order to get a refresh token, you need to authenticate to an application and provide a device identifier. 

For example:

```
{
  "device": "<random_id>",
  "loginId": "example@inversoft.com",
  "password": "password",
  "applicationId": "10000000-0000-0002-0000-000000000001"
}
```

If you authenticate without the Application Id or the device Id you won't receive a Refresh Token because Refresh Tokens are scoped to a user for a device and a specific application.

So you'll need at least one application configured in the Passport Management Interface, or they can also be created via the Application API.

The Access Token (JWT) should always be returned in the LoginResponse. So if you're not receiving an Access Token (the token property in the response) there is something else going on.

**Comment by simone201** [comment]
Here is a quick code snippet that i'm using to log a user, using the app id and such.
https://pastebin.com/ciWQq6qh

The LoginResponse object has always the refreshToken null, even if i set the device...
I hope this clarifies the situation

**Comment by robotdan** [comment]
Thanks, I believe your request looks correct. What is the status code you receive on the response? If you're receiving a `202` that means you're not registered to the application you're providing on the request. If you are not registered for the Application we'll ignore the Application Id that you provide. 

You can register yourself for any application from the Passport Backend Management Interface by Navigating to Users --> Search, searching on your user and select Manage from the row actions. In the Manage panel you can add registrations to your user at the bottom of the screen. 

If you're receiving a `200` that means you're registered and you've been authenticated, and in that case I'd expect you to receive the `token` (JWT) in the response and the `refreshToken` (Refresh Token) in the response if you provided a device. 

Are you running on one of our hosted instances of Passport? If so, I can verify your settings if you'd like.

**Comment by simone201** [comment]
Yeah i'm testing out the free trial for now so it should be in the private cloud, I just found that the administrator account that i was using to log into Passport wasn't registered in the application so it might be this issue (will be trying asap), i thought that by default the account owner can be logger anywhere as happens in many other auth systems.

**Comment by simone201** [comment]
Just tried it out, seems now working fine!
Thanks a lot for the support, maybe it needs a bit more clarity in the documentation and a bit more quickstart guides just to prevent any other requests.

Thanks again!

**Comment by robotdan** [comment]
You're welcome, we do have a few tutorials but we are always working to build out better documentation so thank you for the feedback. 

Had you seen this Register a User & Login Tutorial? ( https://www.inversoft.com/docs/passport/1.x/tech/tutorials/register-user-login-api ) If so, it sounds like there was still some confusion, so I can look at enhancing this tutorial. 

Let us know if you need further assistance.
